cmd/osmosisd/cmd/init.go

```diff
@@ -167,6 +169,18 @@ func InitCmd(mbm module.BasicManager, defaultNodeHome string) *cobra.Command {
 
 			tmcfg.WriteConfigFile(filepath.Join(config.RootDir, "config", "config.toml"), config)
 
+			// create environment file in default node home
+			if defaultNodeHome == app.DefaultNodeHome {
+				envFile, err := os.Create(filepath.Join(defaultNodeHome, ".env"))
```

Comment:
Why are we using `defaultNodeHome` here and not `.osmosisd`?

What if I re-run init command while I'm in another environment?

We need to make sure that the `.env` is always saved in the same place independent of the environment

Reply:
The scenario would be, the first init will initialize the .`osmosisd` folder and the `.env` file (whether you initialize in `defaultNodeHome ` or not). 
From there, the `.env` is only be changed when using `ChangeEnvironmentCmd`. 
If we run-run init, the `.env` still save the last environment